api_bitrise.py

```diff
@@ -0,0 +1,127 @@
+import time
+import sys
+import os
+
+import pandas as pd
+from datetime import datetime
+from datetime import timezone
+from sqlalchemy import func
+
+from lib.bitrise_conn import BitriseAPIClient
+from database import (
+    Database,
+    ReportBitriseBuildsCount
+)
+
+
+class Bitrise:
+
+    def __init__(self):
+        try:
+            BITRISE_HOST = os.environ['BITRISE_HOST']
+            self.client = BitriseAPIClient(BITRISE_HOST)
+            self.client.BITRISE_APP_SLUG = os.environ['BITRISE_APP_SLUG']
+            self.client.token = os.environ['BITRISE_TOKEN']
+        except KeyError:
+            print("ERROR: Missing bitrise env var")
+            sys.exit(1)
+
+    # API: Builds
+    def builds(self, past_date_timestamp):
+        return self.client.get_builds(self.client.BITRISE_APP_SLUG, past_date_timestamp) # noqa
+
+    def get_builds_range_date(self, after):
+        return self.client.get_builds_time(self.client.BITRISE_APP_SLUG, after)
+
+
+class BitriseClient(Bitrise):
+
+    def __init__(self):
+        super().__init__()
+        self.db = DatabaseBitrise()
+
+    def builds_daily_count(self):
+        days_ago = 1
+        today = datetime.datetime.utcnow().date()
+        past_date = today - datetime.timedelta(days=days_ago)
+        print(past_date)
+        past_date_timestamp = int(time.mktime(past_date.timetuple()))
+        print(past_date_timestamp)
+
+        # Pull JSON blob from Bitrise
+        # payload = self.builds(past_date_timestamp)
+
+        # data_frame = self.db.report_bitrise_builds_count(payload)
+        # self.db.report_bitrise_builds_count_insert(data_frame)
+
+    def builds_detailed_info(self):
+        # Read latest timestamp from database
+        after = self.get_latest_build()
+        print(after)
+
+        # Query bitrise using after that date
+        data = self.get_builds_range_date(after)
+
+        payload = pd.DataFrame(data, columns=["build_number", "branch", "status", "status_text", "triggered_workflow", "triggered_by", "triggered_at"]) # noqa
+        payload_filtered = payload[payload["status_text"] != "in-progress"]
+        print(payload_filtered)
+
+        self.db.report_bitrise_builds_info(payload_filtered)
+
+    def get_latest_build(self):
+        # Fetch latest triggered_at
+        latest_ts = self.db.session.query(func.max(ReportBitriseBuildsCount.triggered_at)).scalar() # noqa
```

Comment:
So that I only add new builds, I get the latest `triggered_at` value from database, then I do the query, get all builds after that date, that are not in-progress, that way I am sure I don't miss or duplicate any build. 
Tested it a few times and is working.

Reply:
nice